ui: Pagination page count

vladlos opened the pull request:
Updated the page count on all list views to new format

Resolves: #45262

Release note (ui): none

vladlos committed:
ui: Pagination page count

Updated the page count on all list views to new format

Resolves: #45262

Release note (ui): none

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/45644)
<!-- Reviewable:end -->

dhartunian commented on pkg/ui/src/components/pagination/pagination.tsx:
```diff
@@ -63,3 +65,18 @@ export function PaginationComponent(props: PaginationComponentProps) {
 }
+
+// tslint:disable-next-line: no-shadowed-variable
+export const paginationPageCount = (pagination: PaginationSettings, pageName: string, match?: match<any>, appAttr?: string, search?: string) => {
```
@elkmaster this looks good but can you add some unit tests for this function to demonstrate what it does when page is 0,1, etc. and we have 0,1,2,200 results.

vladlos committed:
ui: Pagination spec

added pagination spec

Release note (ui): none

dhartunian reviewed (changes requested):
Thanks for the tests!

> Reviewed 1 of 3 files at r1, 2 of 2 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/45644#-:-M3gHbtK532oTFsHkcNh:biyn2jc)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @dhartunian and @elkmaster)

___
*[pkg/ui/src/components/pagination/pagination.spec.ts, line 31 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/45644#-M3gHC-I16pZyM0A6Vfw:-M3gHC-I16pZyM0A6Vfx:b6hmlqb) ([raw file](https://github.com/cockroachdb/cockroach/blob/533eeb975289f032f7fb9e1e42e30503c0cf481a/pkg/ui/src/components/pagination/pagination.spec.ts#L31)):*
> ```XML
>       assert.equal(paginationPageCount({ pageSize: 20, current: 1, total: 7 }, "test"), "1-7 of 7 test");
>       assert.equal(paginationPageCount({ pageSize: 20, current: 1, total: 0 }, "test"), "0 test");
>       assert.equal(paginationPageCount({ pageSize: 20, current: 2, total: 300 }, "test", match, "app"), "21-40 of 300 results");
> ```

I have a test case with surprising behavior:

`{ pageSize: 10, current: 2, total: 15 }`

I get
```
Expected :"11-15 of 15 jobs"
Actual   :"6-15 of 15 jobs"
```

I think in this case it's expecting page 2 to be a size of 10 but I would expect page 1 to have size 10 and page 2 to have size 5.


<!-- Sent from Reviewable.io -->

vladlos committed:
ui: Pagination spec

added more pagination tests

Release justification: low risk, high benefit changes to existing functionality

Release note (ui): none

elkmaster commented:
![1](https://user-images.githubusercontent.com/12850886/78565216-3f0b0a80-7826-11ea-9b80-90413e6b7fc4.png)
![2](https://user-images.githubusercontent.com/12850886/78565219-416d6480-7826-11ea-97a6-a57219041fe8.png)
![3](https://user-images.githubusercontent.com/12850886/78565222-4205fb00-7826-11ea-9ed7-b760b8fbb729.png)
![4](https://user-images.githubusercontent.com/12850886/78565224-429e9180-7826-11ea-8d10-ab6318252e08.png)

dhartunian reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 3 of 3 files at r4, 2 of 2 files at r5.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/45644#-:-M4R7D_eDMCvaqmJdKTP:bfsmuih)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @dhartunian and @elkmaster)



<!-- Sent from Reviewable.io -->

dhartunian commented:
bors r+
